services/traction/api/db/models/base.py

```diff
@@ -36,3 +83,78 @@ class BaseTable(BaseModel):
             TIMESTAMP, nullable=False, server_default=func.now(), onupdate=func.now()
         )
     )
+
+
+class TimestampModel(BaseModel):
+    created_at: datetime = Field(
+        sa_column=Column(TIMESTAMP, nullable=False, server_default=func.now())
+    )
+    updated_at: datetime = Field(
+        sa_column=Column(
+            TIMESTAMP, nullable=False, server_default=func.now(), onupdate=func.now()
+        )
+    )
+
+
+class StatefulModel(BaseModel):
+    def check_error_status_detail(context):
+        try:
+            if context.get_current_parameters()["status"] != "Error":
+                return None
+        except builtins.KeyError:
+            pass
```

Comment:
I believe you need to return the value that will get saved, in either case. 

i think you need to return `context.get_current_parameters()["status"]` if status does == error. 

but i'm not sure of that, so if you tested it and it works, then that's great.